File: backend/app/schemas/market.py
from pydantic import BaseModel


class OverviewPointOut(BaseModel):
    t: str
    avg_pct: float | None
    up: int
    limit_up: int
    flat: int
    down: int
    limit_down: int


class OverviewSummaryOut(BaseModel):
    total: int
    with_pre_close: int
    up: int
    limit_up: int
    flat: int
    down: int
    limit_down: int


class OverviewOut(BaseModel):
    trade_date: str
    series: list[OverviewPointOut]
    summary: OverviewSummaryOut


class RankItemOut(BaseModel):
    secucode: str
    name: str
    price: float
    pct: float


class RankingOut(BaseModel):
    time: str
    gainers: list[RankItemOut]
    losers: list[RankItemOut]


class StockMinutePointOut(BaseModel):
    t: str
    price: float
    vol: int
    pct: float | None


class StockMinuteOut(BaseModel):
    secucode: str
    name: str
    pre_close: float | None
    points: list[StockMinutePointOut]

File: frontend/src/api/market.ts
import { apiGet } from "./client";

export interface OverviewPoint {
  t: string;
  avg_pct: number | null;
  up: number;
  limit_up: number;
  flat: number;
  down: number;
  limit_down: number;
}

export interface OverviewSummary {
  total: number;
  with_pre_close: number;
  up: number;
  limit_up: number;
  flat: number;
  down: number;
  limit_down: number;
}

export interface Overview {
  trade_date: string;
  series: OverviewPoint[];
  summary: OverviewSummary;
}

export interface RankItem {
  secucode: string;
  name: string;
  price: number;
  pct: number;
}

export interface Ranking {
  time: string;
  gainers: RankItem[];
  losers: RankItem[];
}

export interface StockMinutePoint {
  t: string;
  price: number;
  vol: number;
  pct: number | null;
}

export interface StockMinute {
  secucode: string;
  name: string;
  pre_close: number | null;
  points: StockMinutePoint[];
}

export const getMarketDates = () => apiGet<string[]>("/market/minute/dates");
export const getMarketOverview = (date: string) =>
  apiGet<Overview>("/market/minute/overview", { date });
export const getMarketRanking = (date: string, time: string) =>
  apiGet<Ranking>("/market/minute/ranking", { date, time });
export const getStockMinute = (date: string, secucode: string) =>
  apiGet<StockMinute>("/market/minute/stock", { date, secucode });
